app/src/test/kotlin/maru/app/MaruAppCliTest.kt

```diff
@@ -8,16 +8,234 @@
+      blocks-parallelism = 10
+      max-retries = 5
+      backoff-delay = "1 seconds"
+      use-unconditional-random-download-peer = false
+      """.trimIndent()
+
+    private val maruConfigOverridesDtoToml =
+      """
+      [persistence]
+      data-path="./tmp/maru-db"
+      private-key-path="./tmp/maru-db/private-key"
+
+      [p2p]
+      port = 3321
+      ip-address = "0.0.0.0"
+      static-peers = []
+      reconnect-delay = "500 ms"
+
+      [follower-engine-apis]
+      "follower-erigon" = { endpoint = "http://localhost:18551", jwt-secret-path = "./tmp/docker/jwt" }
+      "follower-nethermind" = { endpoint = "http://localhost:18552", jwt-secret-path = "./tmp/docker/jwt" }
+      "follower-geth" = { endpoint = "http://localhost:18553", jwt-secret-path = "./tmp/docker/jwt" }
+
+      [payload-validator]
+      engine-api-endpoint = { endpoint = "http://localhost:8550" }
+      eth-api-endpoint = { endpoint = "http://localhost:8545" }
+      """.trimIndent()
+
+    private val expectedMaruConfigDtoToml =
+      """
+      allow-empty-blocks = true
+
+      [persistence]
+      data-path="./tmp/maru-db"
+      private-key-path="./tmp/maru-db/private-key"
+
+      [qbft]
+      fee-recipient = "0x0000000000000000000000000000000000000000"
+
+      [p2p]
+      port = 3321
+      ip-address = "0.0.0.0"
+      static-peers = []
+      reconnect-delay = "500 ms"
+
+      [p2p.discovery]
+      port = 3324
+      bootnodes = ["enr:-Iu4QHk0YN5IRRnufqsWkbO6Tn0iGTx4H_hnyiIEdXDuhIe0KKrxmaECisyvO40mEmmqKLhz_tdIhx2yFBK8XFKhvxABgmlkgnY0gmlwhH8AAAGJc2VjcDI1NmsxoQOgBvD-dv0cX5szOeEsiAMtwxnP1q5CA5toYDrgUyOhV4N0Y3CCJBKDdWRwgiQT"]
+      refresh-interval = "2 minutes"
+
+      [payload-validator]
+      engine-api-endpoint = { endpoint = "http://localhost:8550" }
+      eth-api-endpoint = { endpoint = "http://localhost:8545" }
+
+      [follower-engine-apis]
+      "follower-erigon" = { endpoint = "http://localhost:18551", jwt-secret-path = "./tmp/docker/jwt" }
+      "follower-nethermind" = { endpoint = "http://localhost:18552", jwt-secret-path = "./tmp/docker/jwt" }
+      "follower-geth" = { endpoint = "http://localhost:18553", jwt-secret-path = "./tmp/docker/jwt" }
+
+      [observability]
+      port = 9090
+      jvm-metrics-enabled = true
+      prometheus-metrics-enabled = true
+
+      [api]
+      port = 8080
+
+      [syncing]
+      peer-chain-height-polling-interval = "5 seconds"
+      el-sync-status-refresh-interval = "5 seconds"
+      sync-target-selection = "Highest"
+      # sync-target-selection = { _type = "MostFrequent", peer-chain-height-granularity = 10 }
+      desync-tolerance = 10
+
+      [syncing.download]
+      block-range-request-timeout = "10 seconds"
+      blocks-batch-size = 10
+      blocks-parallelism = 10
+      max-retries = 5
+      backoff-delay = "1 seconds"
+      use-unconditional-random-download-peer = false
+      """.trimIndent()
+
+    private val expectedMaruGenesisJson =
```

Comment:
I feel like this should be a minimal possible parseable genesis file, because its content is not essential for what we're testing. The parsing logic is thoroughly covered in our config tests